pandas/core/arrays/string_.py

```diff
@@ -205,9 +200,9 @@ def __setitem__(self, key, value):
 
         # validate new items
         if scalar_value:
-            if scalar_value is None:
-                value = np.nan
-            elif not (isinstance(value, str) or np.isnan(value)):
+            if isna(value):
```

Comment:
do we want to include _any_ na value?  e.g NaT seems like it should be excluded

Reply:
It's not clear to me. We at least want to handle None, NA, and nan here. You'd prefer raising on NaT?

Reply:
I am personally fine with not allowing pd.NaT (it's clearly specific to time-like data). 

But we should then have a helper function for it (or we probably already have one somewhere), or have it as an option in `isna`, as previously (in the PR to my fork), this was:

```
            if (
                value is libmissing.NA
                or value is None
                or (
                    is_float_dtype(value)
                    and not isinstance(value, str)
                    and np.isnan(value)
                )
            ):
```